Yubico.YubiKey/src/Yubico/YubiKey/Piv/PivSession.ManagementKey.cs

```diff
@@ -1061,5 +1013,68 @@ private bool TryAuthenticateManagementKey(bool mutualAuthentication,
 
             return ManagementKeyAuthenticated;
         }
+
+        private void RefreshManagementKeyAlgorithm() => ManagementKeyAlgorithm = GetManagementKeyAlgorithm();
+
+        private PivAlgorithm GetManagementKeyAlgorithm()
+        {
+            var response = Connection.SendCommand(new GetMetadataCommand(PivSlot.Management));
+            if (response.Status != ResponseStatus.Success)
+            {
+                throw new InvalidOperationException(response.StatusMessage);
```

Comment:
Metadata was added in FW 5.3.0. It is safe to return `PivAlgorithm.TripleDes` here, instead of throwing the exception as YubiKeys with that old FW only support Triple DES; AES management key type was added in 5.4.0.

Reply:
Even though the command can fail for any reason?

Reply:
A more correct approach would be to always return TDES for older FW versions. A non-success status for keys that DO support metadata should still be treated as an error.

Reply:
Nice, thats what I ended up doing